.cirrus.yml

```diff
@@ -185,6 +204,52 @@ task:
         - export CIRRUS_CHANGE_MESSAGE=`cat /tmp/cirrus_change_message.txt`
         - export CIRRUS_COMMIT_MESSAGE=`cat /tmp/cirrus_commit_message.txt`
 
+
+task:
+  use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_PR == ''
+  windows_container:
+    image: cirrusci/android-sdk:28-windowsservercore-2019
+    os_version: 2019
+    cpu: 4
+  env:
+    CIRRUS_WORKING_DIR: "C:\\Windows\\Temp\\flutter sdk"
+  git_fetch_script:
+    - git clean -xfd
+    - git fetch origin
+    - git fetch origin master # To set FETCH_HEAD for "git merge-base" to work
+  pub_cache:
+    folder: $APPDATA\Pub\Cache
+    fingerprint_script:
+      - ps:  $Env:OS; Get-ChildItem -Path "$Env:CIRRUS_WORKING_DIR" pubspec.yaml -Recurse | Select-String -Pattern "PUBSPEC CHECKSUM" -SimpleMatch
+  flutter_pkg_cache:
+    folder: bin\cache\pkg
+    fingerprint_script: echo %OS% & type bin\internal\engine.version
+  artifacts_cache:
+    folder: bin\cache\artifacts
+    fingerprint_script: echo %OS% & type bin\internal\engine.version
+  setup_script:
+    - bin\flutter.bat config --no-analytics
+    - bin\flutter.bat doctor -v
+    - bin\flutter.bat update-packages
+    - git fetch origin master
+  build_runner_cache:
+    fingerprint_script: type bin\cache\dart-sdk\version
+    folder: .\packages\flutter_tools\.dart_tool
+    populate_script: powershell dev\bots\flutter_tool_build_cache.ps1
+  test_all_script:
+    - bin\cache\dart-sdk\bin\dart.exe --enable-asserts dev\bots\test.dart
+  matrix:
+    - name: tool_tests-windows
```

Comment:
What's the different between "tool_tests" and "tool_tests_integration"? Consider adding some comment to clarify.

Reply:
tool_tests runs everything in the `packages/flutter_tools/test` except the integration folder.

tool_tests_integration runs everything in the `packages/flutter_tools/test/integration` folder.  I can add some comments, still trying to see what caching strategy is helping or hurting.